src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -241,7 +241,20 @@ public override void VisitResourceDeclarationSyntax(ResourceDeclarationSyntax sy
 
                 if (singleResourceDeclaredType is ResourceType resourceType && !resourceTypeProvider.HasType(resourceType.TypeReference))
                 {
-                    diagnostics.Write(DiagnosticBuilder.ForPosition(syntax.Type).ResourceTypesUnavailable(resourceType.TypeReference));
+                    var typeSegments = resourceType.TypeReference.Types;
+
+                    if (typeSegments.Length > 2 &&
+                        typeSegments.Where((type, i) => i > 0 && i < (typeSegments.Length - 1) &&  StringComparer.OrdinalIgnoreCase.Equals(type, "providers")).Any())
```

Comment:
>providers [](start = 146, length = 9)

The only nested resource type I could find was `subscriptions/providers` in Microsoft.Resources (searched in the manifest repo). That is read-only, so it should be ok.